x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutTrainedModelVocabularyAction.java

```diff
@@ -71,13 +75,22 @@ public Request(StreamInput in) throws IOException {
             } else {
                 this.merges = List.of();
             }
+            if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_9_0)) {
```

Comment:
In 8.9 `TransportVersion` is an incrementing number, see https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/TransportVersion.java#L34